src/sw.ts

```diff
@@ -2,25 +2,42 @@
 /// <reference lib="webworker" />
 /// <reference no-default-lib="true"/>
 
+import "workbox-core";
+import { offlineFallback, googleFontsCache } from "workbox-recipes";
+import { NavigationRoute, registerRoute } from "workbox-routing";
 import { CONFIG } from "./lib/config";
+import {
+  cleanupOutdatedCaches,
+  createHandlerBoundToURL,
+  precacheAndRoute,
+} from "workbox-precaching";
+
+declare let self: ServiceWorkerGlobalScope;
 
 const OFFLINE_VERSION = CONFIG.RELEASE_VERSION;
 
 console.log("SW VERSION 1", OFFLINE_VERSION);
 
-import { offlineFallback, googleFontsCache } from "workbox-recipes";
-import { setDefaultHandler } from "workbox-routing";
-import { NetworkOnly } from "workbox-strategies";
-
-declare let self: ServiceWorkerGlobalScope;
-
-// Disable workbox logs
-(self as any).__WB_DISABLE_DEV_LOGS = true;
+// Disable workbox logs => do not delete this static import: import "workbox-core";
+self.__WB_DISABLE_DEV_LOGS = true;
 
 self.addEventListener("message", (event) => {
   if (event.data && event.data.type === "SKIP_WAITING") self.skipWaiting();
 });
 
-setDefaultHandler(new NetworkOnly());
+// allow only fallback in dev: we don't want to cache anything in dev
+let allowlist: undefined | RegExp[];
+if (import.meta.env.DEV) allowlist = [/^offline.html$/];
+
+precacheAndRoute(self.__WB_MANIFEST);
+
+// clean old assets
+cleanupOutdatedCaches();
+
 googleFontsCache();
 offlineFallback();
```

Comment:
If the `registerRoute` setup eventually works I can probably remove this as its setting up a separate cache of `offline.html`.

Reply:
Yeah, offlineFallback can be delete from the sw